File: src/Domain/ValueObject/OutageInfo.php
<?php

declare(strict_types=1);

namespace App\Domain\ValueObject;

final readonly class OutageInfo
{
    public function __construct(
        public OutagePeriod $period,
        public OutageDescription $description,
    ) {
    }

    public function equals(self $other): bool
    {
        return $this->period->equals($other->period)
            && $this->description->equals($other->description);
    }
}

File: internal/domain/outage_info.go
package domain

// OutageInfo is a composite value object containing a period and description.
type OutageInfo struct {
	Period      OutagePeriod
	Description OutageDescription
}

// NewOutageInfo creates a new OutageInfo.
func NewOutageInfo(period OutagePeriod, description OutageDescription) OutageInfo {
	return OutageInfo{Period: period, Description: description}
}

// Equals checks if two OutageInfo values are equal.
func (i OutageInfo) Equals(other OutageInfo) bool {
	return i.Period.Equals(other.Period) && i.Description.Equals(other.Description)
}
